### PR TITLE
galleryRP 2024-03-03

### PR DESCRIPTION
**Pull request by grizzlytheodore**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/compute/Azure.ResourceManager.Compute/src/Generated/GalleryImageVersionData.cs [hunk 1]
```diff
@@ -70,15 +70,19 @@ public GalleryImageVersionData(AzureLocation location) : base(location)
...
+        internal GalleryImageVersionData(ResourceIdentifier id, string name, ResourceType resourceType, SystemData systemData, IDictionary<string, string> tags, AzureLocation location, GalleryImageVersionPublishingProfile publishingProfile, GalleryProvisioningState? provisioningState, GalleryImageVersionStorageProfile storageProfile, GalleryImageVersionSafetyProfile safetyProfile, ReplicationStatus replicationStatus, ImageVersionSecurityProfile securityProfile, bool? restore, ValidationsProfile validationsProfile, IDictionary<string, BinaryData> serializedAdditionalRawData) : base(id, name, resourceType, systemData, tags, location)
         {
             PublishingProfile = publishingProfile;
             ProvisioningState = provisioningState;
             StorageProfile = storageProfile;
             SafetyProfile = safetyProfile;
             ReplicationStatus = replicationStatus;
             SecurityProfile = securityProfile;
+            Restore = restore;
```
Comment (archerzz) [inline sdk/compute/Azure.ResourceManager.Compute/src/Generated/GalleryImageVersionData.cs:84]: Consider adding a verb to make the boolean property name more readable, like `EnableRestore`, etc.

File: sdk/compute/Azure.ResourceManager.Compute/src/Generated/Models/EndpointAccess.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.Compute.Models
+{
+    /// <summary> This property allows you to specify if the requests will be allowed to access the host endpoints. Possible values are: 'Allow', 'Deny'. </summary>
+    public readonly partial struct EndpointAccess : IEquatable<EndpointAccess>
```
Comment (archerzz) [inline sdk/compute/Azure.ResourceManager.Compute/src/Generated/Models/EndpointAccess.cs:14]: The name is too generic. Consider appending RP prefix or renaming it.
- use `prepend-rp-prefix` to automatically append PR name. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L52-L57).
- use `rename-mapping` to rename a model. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L58-L76)

File: sdk/compute/Azure.ResourceManager.Compute/src/Generated/Models/EndpointType.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+namespace Azure.ResourceManager.Compute.Models
+{
+    /// <summary> This property allows you to specify the Endpoint type for which this profile is defining the access control for. Possible values are: 'WireServer' or 'IMDS'. </summary>
+    public enum EndpointType
```
Comment (archerzz) [inline sdk/compute/Azure.ResourceManager.Compute/src/Generated/Models/EndpointType.cs:11]: Name is too generic.

File: sdk/compute/Azure.ResourceManager.Compute/src/Generated/Models/GalleryImageVersionPatch.cs [hunk 4]
```diff
@@ -102,6 +106,10 @@ public GalleryImageVersionUefiSettings SecurityUefiSettings
             }
         }
 
+        /// <summary> Indicates if this is a soft-delete resource restoration request. </summary>
+        public bool? Restore { get; set; }
```
Comment (archerzz) [inline sdk/compute/Azure.ResourceManager.Compute/src/Generated/Models/GalleryImageVersionPatch.cs:110]: same, better to use a more readable name

File: sdk/compute/Azure.ResourceManager.Compute/src/Generated/Models/GallerySoftDeletedResource.cs [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        internal GallerySoftDeletedResource()
+        {
+        }
+
+        /// <summary> arm id of the soft-deleted resource. </summary>
+        public ResourceIdentifier ResourceArmId { get; set; }
+        /// <summary> artifact type of the soft-deleted resource. </summary>
+        public SoftDeletedArtifactType? SoftDeletedArtifactType { get; set; }
+        /// <summary> The timestamp for when the resource is soft-deleted. In dateTime offset format. </summary>
+        public string SoftDeletedTime { get; set; }
```
Comment (archerzz) [inline sdk/compute/Azure.ResourceManager.Compute/src/Generated/Models/GallerySoftDeletedResource.cs:85]: Could this be a type of `DateTimeOffset`? See an example [here](https://github.com/Azure/azure-sdk-for-net/blob/e4fce1607d65d709b19c81cc78969cd08b804244/sdk/hdinsight/Azure.ResourceManager.HDInsight/src/autorest.md?plain=1#L146).

File: sdk/compute/Azure.ResourceManager.Compute/src/Generated/Models/PlatformAttribute.cs [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+#nullable disable
+
+using System;
+using System.Collections.Generic;
+
+namespace Azure.ResourceManager.Compute.Models
+{
+    /// <summary> This is the platform attribute of the image version. </summary>
+    public partial class PlatformAttribute
```
Comment (archerzz) [inline sdk/compute/Azure.ResourceManager.Compute/src/Generated/Models/PlatformAttribute.cs:14]: Name is too generic

File: sdk/compute/Azure.ResourceManager.Compute/src/Generated/Models/ValidationStatus.cs [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.Compute.Models
+{
+    /// <summary> This property specifies the status of the validationProfile of the image version. </summary>
+    public readonly partial struct ValidationStatus : IEquatable<ValidationStatus>
```
Comment (archerzz) [inline sdk/compute/Azure.ResourceManager.Compute/src/Generated/Models/ValidationStatus.cs:14]: Name is too generic

**Review by archerzz — DISMISSED** [review]
Please fix the test error: https://dev.azure.com/azure-sdk/public/_build/results?buildId=4385389&view=logs&j=1eb5ce1b-da74-54f5-4ec0-3d620fd997f9&t=cc31fbea-0126-5b56-8b7f-70c253203b81&l=5940

**Comment by grizzlytheodore** [comment]
@ArcturusZhang  your comments are addressed. please take a look again

**Comment by archerzz** [comment]
> @ArcturusZhang your comments are addressed. please take a look again

@grizzlytheodore I think my comments are not resolved?

**Comment by grizzlytheodore** [comment]
> Please fix the test error: https://dev.azure.com/azure-sdk/public/_build/results?buildId=4385389&view=logs&j=1eb5ce1b-da74-54f5-4ec0-3d620fd997f9&t=cc31fbea-0126-5b56-8b7f-70c253203b81&l=5940

I believe test error is resolved by renaming GallerySoftDeletedResource to GallerySoftDeleted

**Comment by grizzlytheodore** [comment]
> > @ArcturusZhang your comments are addressed. please take a look again
> 
> @grizzlytheodore I think my comments are not resolved?

I believe test error is resolved by renaming GallerySoftDeletedResource to GallerySoftDeleted

**Comment by grizzlytheodore** [comment]
please review! @archerzz  @ArcturusZhang

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**